### DISCUSSION TITLE
Upgrade path from KC 18 to 22

### DISCUSSION THREAD

**Discussion by kbrunnerecm** [keycloak/keycloak#21732] [Q&A] [answered]
Dear community

I want to upgrade from KC18 to 22. Is it possible to upgrade directly? I haven't found any information on the upgrade manual.

Thank you in advance.

Best regards
KB

**Comment by bonnm** [comment, marked as answer]
Hi,
Yes, that is possible. I recently did a direct upgrade from KC 18.0.2 (Wildfly) on OpenJDK 11 to KC 21.1.2 on OpenJDK 17. Bare Metal setup on Ubuntu 20, Database is MS SQL Server 2019. Two instances running behind Apache proxy LB in edge mode.

I think the most critical thing in such an upgrade is the database migration, but (on both my test and prod systems) that went without any problems.

After that, I had to adjust some things in Keycloak config und Apache to get the admin console and the sticky sessions work, but that depends on your environment.

Regards,
Matthias

**Reply by dasniko** [reply]
What @bonnm wrote.
Plus, I just published a video on that topic: https://youtu.be/-2E5QSQTGtU
